lib/main.dart

```diff
@@ -40,33 +39,43 @@ class V34 extends StatefulWidget {
 }
 
 class _V34State extends State<V34> {
-  Future<SharedPreferences> _preferences = SharedPreferences.getInstance();
+  PreferencesBloc _preferencesBloc = PreferencesBloc();
 
-  ThemeData _getTheme(AsyncSnapshot<SharedPreferences> snapshot) {
-    bool automatic = snapshot.data.getBool("automatic_dark_theme") ?? false;
-    bool dark = snapshot.data.getBool("dark_theme") ?? false;
-    return AppTheme.getThemeFromPreferences(automatic, dark);
+  @override
+  void initState() {
+    super.initState();
+    _preferencesBloc.add(PreferencesLoadEvent());
+    _preferencesBloc.listen((state) {
+      if (state is PreferencesSavedState) {
+        _preferencesBloc.add(PreferencesLoadEvent());
```

Comment:
I suppose that you trigger the `PreferencesLoadEvent` to be sure that every single object in the app will be refreshed according to the theme?
You can also rename `PreferencesLoadedState ` to `PreferencesUpdatedState ` and `yield` this state just after `PreferencesSavedState` (or even remove this one, as I'm not sure it's useful)